crates/taker/src/routes.rs

```diff
@@ -302,11 +303,9 @@ pub async fn post_withdraw_request(
     Ok(projection::to_mempool_url(txid, *network.inner()))
 }
 
-// TODO: Use non-cookie auth for /metrics endpoint as Prometheus does not
-// support cookie-auth (for now, leave unauthenticated)
 #[rocket::get("/metrics")]
 #[instrument(name = "GET /metrics", skip_all, err)]
-pub async fn get_metrics<'r>() -> Result<String, HttpApiProblem> {
+pub async fn get_metrics<'r>(_basic_auth: BasicAuthGuard) -> Result<String, HttpApiProblem> {
```

Comment:
Is it possible to have either `basic_auth` or `cookie_auth` ? That way we don' thave to login twice if we ever call the endpoint manually.